climate_emotions_map/app.py

```diff
@@ -307,25 +311,18 @@ def update_stacked_bar_plots(
     show_all_responses_checked,
 ):
     """Update the stacked bar plots for all questions based on the selected criteria."""
+    if show_all_responses_checked:
+        threshold = None
+    elif not show_all_responses_checked:
+        threshold = DEFAULT_QUESTION["outcome"]
+
+    figure_lookup_key = (state, is_party_stratify_checked, threshold)
+
     figures = []
     for output in ctx.outputs_list:
         # Example: {'id': {'question': 'q2', 'type': 'stacked-bar-plot'}, 'property': 'figure'}
         question = output["id"]["question"]
-
-        if show_all_responses_checked:
-            threshold = None
-        elif not show_all_responses_checked:
-            threshold = DEFAULT_QUESTION["outcome"]
-
-        figure = make_stacked_bar(
-            question=question,
-            subquestion="all",
-            state=state,
-            stratify=is_party_stratify_checked,
-            threshold=threshold,
-            decimals=NUM_DECIMALS,
-        )
-        figures.append(figure)
+        figures.append(PRERENDERED_BARPLOTS[figure_lookup_key][question])
```

Comment:
Nice! 

There's one other place where we should switch to using the pregenerated figures, which is when setting up the initial content of the domain tabs in `layout.py` :

https://github.com/neurodatascience/us_climate_emotions_map/blob/6771c1a3457db628073cc34953041e526e44267d/climate_emotions_map/layout.py#L443-L449

Let me know if you'd like me to make that change.